### PR TITLE
DT-1023: ✨️ Add in calls to stored procedures to transfer money

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/TransferTransactionDetail.java [hunk 1]
```diff
@@ -19,5 +19,5 @@
     @ApiModelProperty(value = "Credit Transaction")
     private Transaction creditTransaction;
     @ApiModelProperty(value = "Transaction Id")
-    private String transactionId;
+    private Long transactionId;
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/TransferTransactionDetail.java:22]: wrong type original specified

File: src/main/java/uk/gov/justice/hmpps/prison/repository/FinanceRepository.java [hunk 2]
```diff
@@ -10,11 +10,11 @@
     Account getBalances(long bookingId, String agencyId);
 
     void insertIntoOffenderTrans(String prisonId, long offId, long offBookId,
-                                 String subActType, long transPostType, long transNumber,
+                                 String transPostType, String subActType, long transNumber,
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/FinanceRepository.java:13]: parameter corrections

File: src/main/java/uk/gov/justice/hmpps/prison/repository/impl/FinanceRepositoryImpl.java [hunk 3]
```diff
@@ -89,11 +89,11 @@ public void processGlTransNew(final String prisonId, final long offId, final lon
                 .addValue("p_off_book_id", offBookId)
                 .addValue("p_sub_act_type_dr", subActTypeDr)
                 .addValue("p_sub_act_type_cr", subActTypeCr)
-                .addValue("p_payee_pers_id", "")
-                .addValue("p_payee_corp_id", "")
-                .addValue("p_payee_name_text", "")
+                .addValue("p_payee_pers_id", null)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/impl/FinanceRepositoryImpl.java:92]: fields are numeric so not specified should be `null`

File: src/main/java/uk/gov/justice/hmpps/prison/repository/storedprocs/TrustProcs.java [hunk 4]
```diff
@@ -71,7 +72,7 @@ public ProcessGlTransNew(final DataSource dataSource, final NomisV1SQLErrorCodeT
                             new SqlParameter("p_payee_pers_id", Types.NUMERIC),
                             new SqlParameter("p_payee_corp_id", Types.NUMERIC),
                             new SqlParameter("p_payee_name_text", Types.VARCHAR),
-                            new SqlParameter("p_gl_sqnc", Types.NUMERIC),
+                            new SqlInOutParameter("p_gl_sqnc", Types.NUMERIC),
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/storedprocs/TrustProcs.java:75]: spring was having trouble finding the stored procedure, think this helped

File: src/test/java/uk/gov/justice/hmpps/prison/service/FinanceServiceTest.java [hunk 5]
```diff
@@ -72,10 +73,8 @@ void testTransfer_wrongPrison() {
         when(offenderBookingRepository.findByOffenderNomsIdAndActiveFlag(anyString(), anyString())).thenReturn(
                 Optional.of(offenderBooking));
 
-        when(accountCodeRepository.findByCaseLoadTypeAndSubAccountType(anyString(), eq("REG"))).thenReturn(
+        when(accountCodeRepository.findByCaseLoadTypeAndSubAccountType(anyString(), eq("SPND"))).thenReturn(
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/service/FinanceServiceTest.java:76]: found out that the transfer should be between spends and savings

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**